### PR TITLE
Use archtest to avoid unintentional dependencies

### PR DESCRIPTION
**Pull request by matthewmcnew**

### CODE REVIEW

File: pkg/cnb/image_rebaser_test.go [hunk 1]
```diff
@@ -131,7 +131,7 @@ type newRemoteArgs struct {
 func (f *FakeRemoteImageUtilFactory) newRemote(imageName string, baseImage string, secretRef registry.SecretRef) (imgutil.Image, error) {
 	for i, args := range f.args {
 		remoteArgs := newRemoteArgs{ImageName: imageName, BaseImage: baseImage, SecretRef: secretRef}
-		if cmp.DeepEqual(args, remoteArgs)().Success() {
+		if diff := cmp.Diff(args, remoteArgs); diff == "" {
```
Comment (ashwin-venkatesh) [inline pkg/cnb/image_rebaser_test.go]: You were right about `cmp` returning a string afterall

File: pkg/arch_test.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package pkg
+
+import (
+	"testing"
+
+	"github.com/matthewmcnew/archtest"
+)
+
+func TestDependencies(t *testing.T) {
```
Comment (ashwin-venkatesh) [inline pkg/arch_test.go]: does this impact test times in a significant way?

Comment (matthewmcnew) [reply pkg/arch_test.go]: It will only scan direct dependencies so the test time increase is manageable.

Comment (joaopapereira) [reply pkg/arch_test.go]: Looks like the testing time, at least in actions, did not change

**Pull request merged**